File: app/src/main/java/com/nytimes/android/external/playbillingtester/legal/LegalViewHolder.java
package com.nytimes.android.external.playbillingtester.legal;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.nytimes.android.external.playbillingtester.R;

import java.util.Map;

/**
 * it shows the license holder and license value
 */
public class LegalViewHolder extends RecyclerView.ViewHolder {

    @NonNull
    private final TextView title;
    @NonNull
    private final TextView body;

    public LegalViewHolder(@NonNull View itemView) {
        super(itemView);
        title = itemView.findViewById(R.id.legal_item_title);
        body = itemView.findViewById(R.id.legal_item_body);
    }

    public void onBind(@NonNull Map.Entry<String, String> legalText) {
        title.setText(legalText.getKey());
        body.setText(legalText.getValue());
    }
}

File: app/src/main/java/com/nytimes/android/external/register/legal/LegalViewHolder.kt
package com.nytimes.android.external.register.legal

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import com.nytimes.android.external.register.R

/**
 * it shows the license holder and license value
 */
class LegalViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val title: TextView = itemView.findViewById(R.id.legal_item_title)
    private val body: TextView = itemView.findViewById(R.id.legal_item_body)

    fun onBind(legalText: Pair<String, String>) {
        title.text = legalText.first
        body.text = legalText.second
    }
}
